Error building dicom_ismrmrd_utility.cpp

**Describe the bug**

I am building Gadgetron master on a Ubuntu 22.04 machine, from the[ `SIRF-SuperBuild` ](https://github.com/SyneRBI/SIRF-SuperBuild)chain. Gadgetron fails to build because of this line, which seems to have been changed in #1193.
https://github.com/gadgetron/gadgetron/blob/a0318d1349dce1b9799cb9f6b4486c6cfc5e0c7f/gadgets/dicom/dicom_ismrmrd_utility.cpp#L208

The failure arises as warnings get translated into build [errors](https://github.com/gadgetron/gadgetron/blob/a0318d1349dce1b9799cb9f6b4486c6cfc5e0c7f/CMakeLists.txt#L166-L183).

The [error ](https://github.com/SyneRBI/SIRF-SuperBuild/pull/438#issuecomment-1443493867)is 
```
/opt/SIRF-SuperBuild/sources/Gadgetron/gadgets/dicom/dicom_ismrmrd_utility.cpp:208:44: error: format '%lld' expects argument of type 'long long int', but argument 4 has type 'long int' [-Werror=format=]
  208 |                 snprintf(buf, BUFSIZE, "%lld", *study_info.accessionNumber);
      |                                         ~~~^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                            |   |
      |                                            |   long int
      |                                            long long int
      |                                         %ld
cc1plus: all warnings being treated as errors

```

Changing to `snprintf(buf, BUFSIZE, "%ld", *study_info.accessionNumber);` in my build seems to fix it.

I am building ISMRMRD master, [`3cb9cabaa01fad0164eccfb80fe9678471911dbd`](https://github.com/ismrmrd/ismrmrd/commit/3cb9cabaa01fad0164eccfb80fe9678471911dbd).

**Expected behavior**
Build to succeed. 

**Logs/Screenshots**
If applicable, add screenshots or logs to help explain your problem.

**System Information:**
 - OS: Linux, Ubuntu 22.04
 - Compiler: g++ (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
 - Install/deployment method docker image, local compile
 - Version master

**Additional context**
Add any other context about the problem here.

--- a/gadgets/dicom/dicom_ismrmrd_utility.cpp
+++ b/gadgets/dicom/dicom_ismrmrd_utility.cpp
@@ -5,6 +5,8 @@
 
 #include "dicom_ismrmrd_utility.h"
 #include <stdio.h>
+#include <cstdio>
+#include <cinttypes>
 #include "boost/date_time/gregorian/gregorian.hpp"
 
 namespace Gadgetron
@@ -205,7 +207,7 @@ namespace Gadgetron
             // Accession Number
             key.set(0x0008, 0x0050);
             if (study_info.accessionNumber) {
-                snprintf(buf, BUFSIZE, "%ld", *study_info.accessionNumber);
+                snprintf(buf, BUFSIZE, "%+" PRId64, *study_info.accessionNumber);
                 write_dcm_string(dataset, key, buf);
             }
             else {

--- a/gadgets/dicom/dicom_ismrmrd_utility.cpp
+++ b/gadgets/dicom/dicom_ismrmrd_utility.cpp
@@ -205,7 +205,7 @@ namespace Gadgetron
             // Accession Number
             key.set(0x0008, 0x0050);
             if (study_info.accessionNumber) {
-                snprintf(buf, BUFSIZE, "%lld", *study_info.accessionNumber);
+                snprintf(buf, BUFSIZE, "%ld", *study_info.accessionNumber);
                 write_dcm_string(dataset, key, buf);
             }
             else {

